src/freertos_drivers/ti/CC32xxUart.cxx

```diff
@@ -159,50 +192,30 @@ void CC32xxUart::interrupt_handler()
             unsigned char c = data;
             if (rxBuf->put(&c, 1) == 0)
             {
-                overrunCount++;
+                ++overrunCount;
             }
             rxBuf->signal_condition_from_isr();
         }
     }
     /* transmit a character if we have pending tx data */
-    if (txPending)
+    if (txPending && (status & UART_INT_TX))
     {
-        while (MAP_UARTSpaceAvail(base))
+        if (txBuf->pending())
+        {
+            send();
+            txBuf->signal_condition_from_isr();
+        }
+        else
         {
-#if 0           
             if (MAP_UARTTxIntModeGet(base) == UART_TXINT_MODE_EOT)
```

Comment:
why would this condition be false?

it seems there are a couple of internal assumptions for the uart driver, I wonder if we could collect those into a consistency check (a set of assertions basically). It might help finding the cause of crashes later.